### DISCUSSION TITLE
error NU1605: Warning As Error: Detected package downgrade: Avalonia from 11.0.0-rc1.1 to 11.0.0-preview4. Reference the package directly from the project to select a different version.

### DISCUSSION THREAD

**Discussion by jim-jiang-github** [AvaloniaUI/Avalonia#11753] [Q&A] [answered]
I have no problem executing publish locally, but the same command will report an error when executed in github action: Avalonia from 11.0.0-rc1.1 to 11.0.0-preview4. Reference the package directly from the project to select a different version.
Why is that?
I have unified all avalonia versions in preview4:
![image](https://github.com/AvaloniaUI/Avalonia/assets/29189410/94c3af07-9a61-4082-90cd-f6f308dc2a53)

This is my github CI: https://github.com/jim-jiang-github/LogVisualizer/actions/runs/5243124158/jobs/9467399537
![image](https://github.com/AvaloniaUI/Avalonia/assets/29189410/0253185c-ceaa-4dd7-a48c-727be1e26c00)

**Comment by timunie** [comment]
looks like one of your packages or any 3rd party lib is referencing another Avalonia version. As we have breaking changes between 11.0 previews, this is not possible. Please check your dependencies for the same version as you use, so they must support 11.0 RC1 in your case.

**Reply by timunie** [reply]
Let me add: preview4 is pretty outdated now, use RC1 is highly reconmented

**Reply by jim-jiang-github** [reply]
> looks like one of your packages or any 3rd party lib is referencing another Avalonia version. As we have breaking changes between 11.0 previews, this is not possible. Please check your dependencies for the same version as you use, so they must support 11.0 RC1 in your case.

I checked the code again and didn't seem to find 3rd party lib referencing another Avalonia version. It's puzzling.

I had to use the preview4 version because there were performance issues with the listbox in the new version of Avalonia, then the core function of my program is to display a lot of data on the listbox.

**Comment by rabbitism** [comment, marked as answer]
Your CI runs a submodule update, which upgrades the MessageBox.Avalonia submodule to rc1 commit. 
![image](https://github.com/AvaloniaUI/Avalonia/assets/14807942/4d51be0e-7e9d-4942-a22c-7dfee06f06e2)

**Reply by rabbitism** [reply]
look at the second step in my screenshot, the submodule is already upgraded to the latest commit in your ci build.

**Reply by jim-jiang-github** [reply]
> look at the second step in my screenshot, the submodule is already upgraded to the latest commit in your ci build.

I see, I see. thank you very very very much!!

**Reply by jim-jiang-github** [reply]
> look at the second step in my screenshot, the submodule is already upgraded to the latest commit in your ci build.

Thank you very much, it has now been successfully compiled
